### PR TITLE
🧪 Add fine grain optimize benchmark

### PR DESCRIPTION
**Pull request by s-weigand**
Adds additional benchmarks based on the test in `glotaran/analysis/test/test_optimization.py`

### Change summary

- `asv` benchmarks different paths for `optimize`

### Checklist

- [x] ✔️ Passing the tests (mandatory for all PR's)

### CODE REVIEW

File: benchmark/benchmarks/unit/analysis/bench_optimize.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            self.scheme.group = grouped
+
+    def time_optimize(self, index_dependent, grouped, weight):
+        optimize(self.scheme)
+
+
+if __name__ == "__main__":
+    bench = BenchmarkOptimize()
+    bench.setup(True, True, True)
+    bench.time_optimize(True, True, True)
```
Comment (jsnel) [inline benchmark/benchmarks/unit/analysis/bench_optimize.py:85]: Is this left here intentionally?

Comment (s-weigand) [reply benchmark/benchmarks/unit/analysis/bench_optimize.py]: Debug artifact thanks for catching it 😄

**Review by jsnel — APPROVED** [review]
Looks good! Much improved visualization of the output results.

**Comment by s-weigand** [comment]
closes #721

**Pull request merged**